src/charm.py

```diff
@@ -132,18 +132,7 @@ def _configure(self, _):
 
         # Restart prometheus only if command line arguments have changed,
         # otherwise just reload its configuration.
-        if current_services == new_layer.services:
-            external_url = urlparse(self._external_url)
-
-            prometheus_server = Prometheus(web_route_prefix=external_url.path)
-
-            reloaded = prometheus_server.reload_configuration()
-            if not reloaded:
-                logger.error("Prometheus failed to reload the configuration")
-                self.unit.status = BlockedStatus(CORRUPT_PROMETHEUS_CONFIG_MESSAGE)
-                return
-            logger.info("Prometheus configuration reloaded")
-        else:
+        if current_services != new_layer.services:
```

Comment:
When the pebble plan has changed, which should happen primarily due changes in the command line arguments, then shouldn't we be restarting Prometheus rather than reloading its configuration.